lib/api/core/models/security/profile.js

```diff
@@ -55,46 +82,48 @@ Profile.prototype.validateDefinition = function () {
  *
  * @return {Promise}
  */
-Profile.prototype.getRights = function() {
+Profile.prototype.getRights = function(kuzzle) {
   var profileRights = {};
 
-  this.roles.forEach(role => {
-    Object.keys(role.controllers).forEach(controller => {
-      var restrictedTo = _.cloneDeep(role.restrictedTo);
-
-      Object.keys(role.controllers[controller].actions).forEach(action => {
-        var actionRights = role.controllers[controller].actions[action];
-
-        if (restrictedTo === undefined || restrictedTo.length === 0) {
-          restrictedTo = [{index: '*', collections: ['*']}];
-        }
-
-        restrictedTo.forEach(restriction => {
-          if (restriction.collections === undefined || restriction.collections.length === 0) {
-            restriction.collections = ['*'];
-          }
-
-          restriction.collections.forEach(collection => {
-            var
-              rightsObject = {},
-              rightsItem = {
-                controller: controller,
-                action: action,
-                index: restriction.index,
-                collection: collection
-              },
-              rightsKey = md5(JSON.stringify(rightsItem));
-
-            rightsItem.value = actionRights;
-            rightsObject[rightsKey] = rightsItem;
-            profileRights = _.assignWith(profileRights, rightsObject, mergeRights);
+  return this.getRoles(kuzzle)
+    .then(roles => {
+      roles.forEach(role => {
+        Object.keys(role.controllers).forEach(controller => {
+          var restrictedTo = _.cloneDeep(role.restrictedTo);
+
+          Object.keys(role.controllers[controller].actions).forEach(action => {
+            var actionRights = role.controllers[controller].actions[action];
+
+            if (restrictedTo === undefined || restrictedTo.length === 0) {
+              restrictedTo = [{index: '*', collections: ['*']}];
+            }
+
+            restrictedTo.forEach(restriction => {
+              if (restriction.collections === undefined || restriction.collections.length === 0) {
+                restriction.collections = ['*'];
+              }
+
+              restriction.collections.forEach(collection => {
+                var
+                  rightsObject = {},
+                  rightsItem = {
+                    controller: controller,
+                    action: action,
+                    index: restriction.index,
+                    collection: collection
+                  },
+                  rightsKey = md5(JSON.stringify(rightsItem));
+
+                rightsItem.value = actionRights;
+                rightsObject[rightsKey] = rightsItem;
+                profileRights = _.assignWith(profileRights, rightsObject, mergeRights);
```

Comment:
No need to assign the result to the same variable, as `_.assignWith` mutates the target object:
`_.assignWith(profileRights, rightsObject, mergeRights);`